install/kubernetes/helm/istio/charts/mixer/templates/deployment.yaml

```diff
@@ -1,88 +1,77 @@
+{{- define "mixer_containers" }}
+    spec:
+      serviceAccountName: istio-mixer-service-account
+      volumes:
+      - name: istio-certs
+        secret:
+          secretName: istio.istio-mixer-service-account
+          optional: true
+    {{- if $.Values.nodeSelector }}
+      nodeSelector:
+{{ toYaml $.Values.nodeSelector | indent 8 }}
+    {{- end }}
+      containers:
+      - name: mixer
+        image: "{{ $.Values.global.hub }}/mixer:{{ $.Values.global.tag }}"
+        imagePullPolicy: {{ $.Values.imagePullPolicy }}
+        ports:
+        - containerPort: 9091
+        - containerPort: 9093
+        - containerPort: 42422
+        args:
+          - --configStoreURL=k8s://
+          - --configDefaultNamespace={{ $.Release.Namespace }}
+          - --trace_zipkin_url=http://zipkin:9411/api/v1/spans
+        resources:
+{{ toYaml $.Values.resources | indent 12 }}
+      - name: istio-proxy
+        image: "{{ $.Values.global.hub }}/{{ $.Values.global.proxy.image }}:{{ $.Values.global.tag }}"
+        imagePullPolicy: {{ $.Values.global.imagePullPolicy }}
+        ports:
+        - containerPort: 15004
+        args:
+        - proxy
+        - mixer
+      {{- if $.Values.global.controlPlaneSecurityEnabled }}
+        - --controlPlaneAuthPolicy
+        - MUTUAL_TLS
+        - --customConfigFile
+        - /etc/istio/proxy/envoy_mixer_auth.json
+      {{- else }}
+        - --controlPlaneAuthPolicy
+        - NONE
+        - --customConfigFile
+        - /etc/istio/proxy/envoy_mixer.json
+      {{- end }}
+        resources:
+{{ toYaml $.Values.global.proxy.resources | indent 12 }}
+        volumeMounts:
+        - name: istio-certs
+          mountPath: /etc/certs
+          readOnly: true
+{{- end }}
+
+
+{{-  $mixers := list "policy" "telemetry" }}
+{{- range $idx, $mname := $mixers }}
 apiVersion: extensions/v1beta1
 kind: Deployment
 metadata:
-  name: istio-mixer
+  name: istio-{{ $mname }}
   labels:
-    app: {{ template "mixer.name" . }}
-    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
-    release: {{ .Release.Name }}
-    heritage: {{ .Release.Service }}
+    chart: {{ $.Chart.Name }}-{{ $.Chart.Version | replace "+" "_" }}
```

Comment:
why the `$` throughout this changeset?

Reply:
If I don't use it I get this error
```Error: render error in "istio/charts/mixer/templates/deployment.yaml": template: istio/charts/mixer/templates/deployment.yaml:62:20: executing "istio/charts/mixer/templates/deployment.yaml" at <.Chart.Name>: can't evaluate field Chart in type interface {}```

This is because that code is in a loop.

Reply:
thanks makes sense.